samples/matter/lock/README.rst

```diff
@@ -377,6 +377,35 @@ To test switching between Thread and Wi-Fi, complete the following steps:
 
       D: 823 [DL]WiFiManager has been initialized
 
+Device firmware upgrade over Matter with support for switching between Thread and Wi-Fi
+==================================
+
+To upgrade the device firmware, complete the steps from the *Device Firmware Upgrade over Matter* section in the :doc:`matter:nrfconnect_examples_software_update` tutorial of the Matter documentation.
+In the case of an application which supports switching between Thread and Wi-Fi you need to generate proper Matter OTA image file which needs to be advertised by Matter OTA Provider.
```

Comment:
```suggestion
Because the application supports switching between Thread and Wi-Fi, you need to make sure that the Matter OTA image file served by the OTA Provider includes two application variants: for Matter over Thread and Matter over Wi-Fi.
```